tap_quickbooks/quickbooks/reportstreams/ARAgingSummaryReport.py

```diff
@@ -39,56 +39,65 @@ def sync(self, catalog_entry):
+            # Recursively get row data.
+            row_group = resp.get("Rows")
+            row_array = row_group.get("Row")
 
-        if row_array is None:
-            return
+            if row_array is None:
+                return
 
-        output = []
-        for row in row_array:
-            if "Header" in row:
-                output.append([i.get('value') for i in row.get("Header", {}).get("ColData", [])])
+            output = []
+            for row in row_array:
+                if "Header" in row:
+                    output.append([i.get('value') for i in row.get("Header", {}).get("ColData", [])])
 
-                for subrow in row.get("Rows", {}).get("Row", []):
-                    output.append([i.get('value') for i in subrow.get("ColData", [])])
+                    for subrow in row.get("Rows", {}).get("Row", []):
+                        output.append([i.get('value') for i in subrow.get("ColData", [])])
 
-                output.append([i.get('value') for i in row.get("Summary", {}).get("ColData", [])])
-            elif "Summary" in row:
-                output.append([i.get('value') for i in row.get("Summary", {}).get("ColData", [])])
-            else:
-                output.append([i.get('value') for i in row.get("ColData", [])])
-
-        # Zip columns and row data.
-        for raw_row in output:
-            row = dict(zip(columns, raw_row))
-            if not row.get("Total"):
-                # If a row is missing the amount, skip it
-                continue
-
-            cleansed_row = {}
-            for k, v in row.items():
-                if v == "":
-                    continue
+                    output.append([i.get('value') for i in row.get("Summary", {}).get("ColData", [])])
+                elif "Summary" in row:
+                    output.append([i.get('value') for i in row.get("Summary", {}).get("ColData", [])])
                 else:
-                    cleansed_row.update({k: v})
-
-            cleansed_row["SyncTimestampUtc"] = singer.utils.strftime(singer.utils.now(), "%Y-%m-%dT%H:%M:%SZ")
-
-            yield cleansed_row
+                    output.append([i.get('value') for i in row.get("ColData", [])])
+
+            # Zip columns and row data.
+            for raw_row in output:
+                row = dict(zip(columns, raw_row))
+                if report_date:
+                    row["report_date"] = report_date
+                if not row.get("Total"):
+                    # If a row is missing the amount, skip it
+                    continue
+                
+                cleansed_row = {}
+                for k, v in row.items():
+                    if v == "":
+                        continue
+                    else:
+                        cleansed_row.update({k: v})
+                cleansed_row["SyncTimestampUtc"] = singer.utils.strftime(singer.utils.now(), "%Y-%m-%dT%H:%M:%SZ")
+                all_rows.append(cleansed_row)
```

Comment:
I think we could keep the `yield cleansed_row` for memory purposes in case of large data

Reply:
Got it. Updated.